Create gitcoin-bounty-issue.md

reachjason opened the pull request:
Adding a template for people to propose bounties that would later go up on Gitcoin.

### Description
Adding a template for people to propose bounties that would later go up on Gitcoin. Users will see a UI on clicking New Issue that looks like the screenshot below:
![image](https://user-images.githubusercontent.com/23431890/111364486-4afa8000-86b7-11eb-8e84-7525b0aecb3a.png)

### Tested
Tested in another repo.

### Backwards compatibility
These changes are backward compatible.

reachjason committed:
Create gitcoin-bounty-issue.md

Adding a template for people to propose bounties that would later go up on Gitcoin.

barbaraliau committed:
Merge branch 'master' into patch-6

barbaraliau commented on .github/ISSUE_TEMPLATE/gitcoin-bounty-issue.md:
```diff
@@ -0,0 +1,21 @@
+name: New Gitcoin Bounty Issue
+about: Use this template to propose new Gitcoin bounties.
+title: "BOUNTY: [Bounty Title]"
+labels: bounty
```
Changing this to `dog` since we've been using the `bounty` label only when it's been approved.

barbaraliau committed:
Update gitcoin-bounty-issue.md

Add additional comments, prerequisites section. Update labels and title formatting.

barbaraliau committed:
Merge branch 'master' into patch-6

barbaraliau reviewed (approved):
TIL you can do comments that will not be rendered in the HTML output https://stackoverflow.com/questions/4823468/comments-in-markdown
